File: cmd/kill_all.go
package cmd

import (
	"context"
	"fmt"
	"os"

	"portkill/internal/finder"
	"portkill/internal/formatter"
	"portkill/internal/killer"
	"portkill/internal/port"

	"github.com/spf13/cobra"
)

func init() {
	rootCmd.AddCommand(killAllCmd)
}

var killAllCmd = &cobra.Command{
	Use:   "kill-all <port>...",
	Short: "Kill processes on multiple ports",
	Args:  cobra.MinimumNArgs(1),
	RunE: func(cmd *cobra.Command, args []string) error {
		ports, err := port.ParsePorts(args[0])
		if err != nil {
			return fmt.Errorf("invalid port: %w", err)
		}

		ctx := context.Background()
		var allProcesses []finder.ProcessInfo

		for _, p := range ports {
			processes, err := finder.FindPort(ctx, p)
			if err != nil {
				fmt.Fprintf(os.Stderr, "Failed to find processes on port %d: %v\n", p, err)
				continue
			}
			allProcesses = append(allProcesses, processes...)
		}

		if len(allProcesses) == 0 {
			fmt.Println("No processes found")
			return nil
		}

		formatter.PrintProcesses(allProcesses, flagVerbose)

		if !flagYes && !formatter.Confirm(allProcesses) {
			fmt.Println("Aborted")
			return nil
		}

		k := killer.NewKiller()
		var lastErr error
		for _, p := range allProcesses {
			if err := k.Kill(ctx, p.PID, flagForce); err != nil {
				fmt.Fprintf(os.Stderr, "Failed to kill PID %d: %v\n", p.PID, err)
				lastErr = err
			} else {
				fmt.Printf("Killed PID %d (%s) on port %d\n", p.PID, p.Name, p.Port)
			}
		}

		return lastErr
	},
}

File: src/commands/killAll.js
const { Command } = require('commander');
const { createFinder, createKiller } = require('../index');
const { printProcesses, confirm, parsePorts, styles } = require('../utils/formatter');

const killAllCmd = new Command('kill-all');
killAllCmd
  .description('Kill processes on multiple ports')
  .argument('<ports>', 'Comma-separated port numbers')
  .action(async (portsStr) => {
    const ports = parsePorts(portsStr);
    const options = killAllCmd.parent.opts();
    const force = options.force || false;
    const verbose = options.verbose || false;
    const yes = options.yes || false;

    const finder = createFinder();
    const killer = createKiller();

    const allProcesses = [];

    for (const port of ports) {
      try {
        const processes = await finder.findByPort(port);
        allProcesses.push(...processes);
      } catch (err) {
        console.error(`${styles.warning('Warning:')} Failed to find processes on port ${port}: ${err.message}`);
      }
    }

    if (allProcesses.length === 0) {
      console.log(styles.muted('No processes found'));
      return;
    }

    printProcesses(allProcesses, verbose);

    if (!yes) {
      const confirmed = await confirm(allProcesses);
      if (!confirmed) {
        console.log(styles.muted('Aborted'));
        return;
      }
    }

    let hasError = false;
    for (const p of allProcesses) {
      try {
        await killer.kill(p.pid, force);
        console.log(styles.success(`✓ Killed ${p.name} (PID: ${p.pid}) on port ${p.port}`));
      } catch (err) {
        console.error(styles.error(`✗ Failed to kill PID ${p.pid}: ${err.message}`));
        hasError = true;
      }
    }

    if (hasError) {
      process.exit(1);
    }
  });

module.exports = killAllCmd;
